### PR TITLE
fix: several issues

### LINKED ISSUE #74: In reports, closed issues should be reported as "closed" not "close" :)

### PR DESCRIPTION
**Pull request by jacobdotcosta**
fix several issues:
* ordering of issue status
* split time is 12:00 on thursday and not 00:00
* some repositories were not being taken into account
* #74 - wrong closed label

Closes #74

### CODE REVIEW

**Comment by metacosm** [comment]
Please try to keep commits atomic: one commit per atomic issue being addressed instead of fixing several things at once in a given commit.

**Comment by iocanel** [comment]
> Please try to keep commits atomic: one commit per atomic issue being addressed instead of fixing several things at once in a given commit.

+1

- It's often really hard not to fix things if you come by them. If you can't resist, you can still stage it as part of a different commit.

- Also, we generally prefer `rebasing` over `merging` as its produces a cleaner history (personally I find merge commits really confusing): https://www.atlassian.com/git/tutorials/merging-vs-rebasing.

File: src/main/java/io/snowdrop/reporting/model/Issue.java [hunk 1]
```diff
@@ -40,42 +40,56 @@
...
-    createdAt = pCreatedAt;
-    updatedAt = pUpdatedAt;
-    closedAt = pClosedAt;
-    source = pSource;
-    status = pStatus;
-    label = pLabel;
-    issueGroup = pIssueGroup;
-    affectedVersion = pAffectedVersion;
-    fixVersion = pFixVersion;
+  String url,
```
Comment (iocanel) [inline src/main/java/io/snowdrop/reporting/model/Issue.java:60]: Yay!

**Review by iocanel — APPROVED** [review]
LGTM

**Comment by jacobdotcosta** [comment]
> > Please try to keep commits atomic: one commit per atomic issue being addressed instead of fixing several things at once in a given commit.
> 
> +1
> 
> * It's often really hard not to fix things if you come by them. If you can't resist, you can still stage it as part of a different commit.
> * Also, we generally prefer `rebasing` over `merging` as its produces a cleaner history (personally I find merge commits really confusing): https://www.atlassian.com/git/tutorials/merging-vs-rebasing.

I've red the article and I think I understand the concept. @iocanel  By rebasing in this situation  wouldn't I be breaking **The Golden Rule of Rebasing** as I'd be rebasing to the master?

**Comment by metacosm** [comment]
> > * It's often really hard not to fix things if you come by them. If you can't resist, you can still stage it as part of a different commit.

Depending on your idea, you can also commit only chunks of your changes at a time. It's also a good practice to work on something and commit as soon as you are done with a given change.

> > * Also, we generally prefer `rebasing` over `merging` as its produces a cleaner history (personally I find merge commits really confusing): https://www.atlassian.com/git/tutorials/merging-vs-rebasing.
> 
> I've red the article and I think I understand the concept. @iocanel By rebasing in this situation wouldn't I be breaking **The Golden Rule of Rebasing** as I'd be rebasing to the master?

There are lots of differing opinions on the subject and I'm not sure what the Golden Rule of Rebasing is, to be honest. If that rule states that you shouldn't rebase master (i.e. rewrite master's history) then it should definitely be avoided. 

However, that's not what occurs when we rebase and merge: in this case, your branch commits are applied to master after all the existing commits (assuming no conflicts exist). That way, history is preserved, i.e. we don't change master from underneath users' feet who might have checked it out locally, we're just adding new commits in a linear fashion instead of creating a merge commit. 

There are merits for both approaches. I also prefer the rebase & merge approach to the merge commit approach but some people think creating merge commits is better.

**Comment by iocanel** [comment]
@jacobdotcosta What works best for most of open source projects I've seen is the following:

- Always work on a different branch
- When you create the branch make sure you rebase: `git pull --rebase origin master` (so that you work against the latest master)
- Whenever you need the latest things from master do a `git pull --rebase origin master` 
- When ready create a pull request (usually its a  good idea to rebase before the pull request).
- If you have multiple commits you want to combine into one (git rebase -i HEAD~<number of commits>).

This way you never interfere with the master directly.
Even if you do, rebase will bump your changes at the end of the commit chain and will not affect the actual history.

**Comment by jacobdotcosta** [comment]
Thanks for taking your time into educating me @metacosm @iocanel . I did merge by default, didn't know better, but I'll try to put into practice these best practices and concepts.

**Pull request merged**